packages/runtime/test/execute/expression.test.ts

```diff
@@ -139,17 +140,20 @@ test.serial('async operations run in series', async (t) => {
   t.is(result.data.x, 12);
 });
 
-test.serial('jobs can return undefined', async (t) => {
-  // @ts-ignore violating the operation contract here
-  const job = [() => undefined] as Operation[];
+test.serial(
+  'jobs return null-state instead of undefined or null',
+  async (t) => {
+    // @ts-ignore violating the operation contract here
+    const job = [() => undefined] as Operation[];
 
-  const state = createState() as TestState;
-  const context = createContext();
+    const state = createState() as TestState;
+    const context = createContext();
 
-  const result = (await execute(context, job, state, {})) as TestState;
+    const result = (await execute(context, job, state, {})) as TestState;
 
-  t.assert(result === undefined);
-});
+    t.assert(isNullState(result));
```

Comment:
function `isNullState` does an object check. 
```js
export function isNullState(state: any) {
  return typeof state === 'object' && state[NULL_STATE] === true;
}
```
Because nullState is also an object.